### ISSUE TITLE
OBS Occasionally crashes with "The Wayland connection broke. Did the Wayland compositor die?"

### ISSUE THREAD

**Issue by Baa14453** [obsproject/obs-studio#6990]
### Operating System Info

Ubuntu 22.04

### OBS Studio Version

27.2.4

### OBS Studio Log URL

Pending

### Expected Behavior

OBS would continue as normal.

### Current Behavior

Occasionally while OBS is running and showing the VLC Video Source it will suddenly crash. Sometimes this happens specifically when I focus the window after it's been behind another program. Sometimes it happens through no prompt, or maybe there is one but I don't notice.

### Steps to Reproduce

1. Open OBS
2. Create a VLC Video Source
3. Leave it playing for up to 4 hours.
4. At some point OBS will crash, the terminal will show this message:

```
The Wayland connection broke. Did the Wayland compositor die?
Aborted (core dumped)
```

### Anything else we should know?

There's too much info revealed in the log from this crash so I'm just replicating it again with less scenes active. I'll upload soon.

I'm using Wayland, recently upgraded to Ubuntu 22.04.1 from Ubuntu 20.04.1.

I originally experienced this with the Snap install but it also replicates with the apt repository.

**Comment by eric-burel** [comment]
Same issue, extremely problematic as it happens to me after just 10 minutes of recording...

**Comment by barf** [comment]
Can also reproduce this, makes OBS unusable unfortunately

Recording/streaming not required to reproduce, apparently occurs with one instance in preview only, after some time switching windows or normal desktop use (issue with new Wayland version in Ubuntu 22.04? Was flawless before upgrade)

**Comment by abcdw** [comment]
It maybe related to #4295.

**Comment by chronos000** [comment]
I can reproduce this issue on my system. I only need to click on any drop-down menu (to open it), and then click enywhere else in the application to cause wayland to crash immediately, with the following logs being generated:

```
kernel: gnome-shell[3317]: segfault at 20 ip 00007fc4e1937639 sp 00007fff7a672d78 error 4 in libmutter-10.so.0.0.0[7fc4e1849000+143000] likely on CPU 7 (core 3, socket 0)
kernel: Code: f0 48 83 c4 08 5b 5d c3 66 0f 1f 84 00 00 00 00 00 48 83 7b 20 00 75 b9 48 8b 5b 08 48 85 db 75 f0 48 83 c4 08 31 c0 5b 5d c3 <48> 8b 04 25 20 00 00 00 0f 0b b8 01 00 00 00 c3 b8 01 00 00 00 eb
com.obsproject.Studio.desktop[281801]: The Wayland connection broke. Did the Wayland compositor die?
```
My system is a PopOS! 22.04, running gnome/mutter, and wayland. I installed OBS via flatpak (currently installed version: 29.1.3)

I also reproduced this crash on a virtual machine by running a live PopOS 22.04 (pop-os_22.04_amd64_intel_22.iso). 

Steps to reproduce:
- boot any computer with pop-os_22.04_amd64_intel_22.iso
- install wayland via flatpak [flatpak install com.obsproject.Studio]
- follow this steps to change the current desktop session to Wayland without rebooting: https://askubuntu.com/questions/1267779/try-wayland-on-20-04-live-usb
- run OBS Studio
- click on any drop-down menu (for example, the one showing "Fade" under "Scene Transitions" . Then click outside of it (like of you were trying to close the drop-down menu without selecting any option)

This will crash wayland and you will loose the session completely, pushing you back to the login screen.

I can run some version of OBS on my system with debuging enabled, and provide the results if anybody lets me know what to provide.

**Comment by abcdw** [comment]
For me it was a qt bug https://bugreports.qt.io/browse/QTBUG-81504, just fixed qt in the distro: https://git.savannah.gnu.org/cgit/guix.git/commit/?id=a43c524252

**Comment by chronos000** [comment]
Thank you @abcdw for submitting that patch. I can see you submitted the patch in January 2023, but I can't find for which version of QT this patch was merged.

In my case, I installed OBS Studio via Flatpak (one of the two official methods), using the standard repository flathub.

I understand that the QT framework is available when Flatpak installs the runtime required by the application.

Here you can see the KDE runtime version installed by Flatpak for OBS Studio:
```
~$ flatpak info --verbose --show-extensions com.obsproject.Studio

OBS Studio - Live streaming and video recording software

          ID: com.obsproject.Studio
         Ref: app/com.obsproject.Studio/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 29.1.3
     License: GPL-2.0-or-later
      Origin: flathub
  Collection: org.flathub.Stable
Installation: user
   Installed: 520.2 MB
     Runtime: org.kde.Platform/x86_64/6.4
         Sdk: org.kde.Sdk/x86_64/6.4

      Commit: e045c06e1a995b02c92612a4fb87ea4830cfb22e38bbb2d94b5ef4d5ce02d799
      Parent: a2a07309102914200525da7264170751b657f635fa203213f08d4c5868d3842f
     Subject: Export com.obsproject.Studio
        Date: 2023-06-19 10:04:09 +0000
F: No installations directory in /etc/flatpak/installations.d. Skipping

   Extension: runtime/com.obsproject.Studio.Locale/x86_64/stable
          ID: com.obsproject.Studio.Locale
      Origin: flathub
      Commit: 28920044196415a40f6a13f83bdfd60d991c3fbceb59e1f5e08ead08e393fdce
   Installed: <512 bytes
    Subpaths: /en

```

Perhaps your patch was merged for a future version of KDE/QT ?

**Comment by abcdw** [comment]
@chronos000 We applied this patch independently in guix package manager, I don't know if it was backported to any 5.x version of upstream qt.

The fix itself is done by another person, not me.

**Comment by GeorgesStavracas** [comment]
@chronos000 the issue you're reporting in your comment is not what this issue is about. This issue is about OBS crashing, your issue is about Mutter / GNOME Shell crashing. Please open an issue in https://gitlab.gnome.org/GNOME/mutter/ instead.

**Comment by chronos000** [comment]
> @chronos000 the issue you're reporting in your comment is not what this issue is about. This issue is about OBS crashing, your issue is about Mutter / GNOME Shell crashing. Please open an issue in https://gitlab.gnome.org/GNOME/mutter/ instead.

Thank you for the heads-up. I have submitted a bug report here:
https://gitlab.gnome.org/GNOME/mutter/-/issues/2966

**Comment by RytoEX** [comment]
https://bugreports.qt.io/browse/QTBUG-81504 was fixed in Qt 6.2.1 and 6.3.0.
https://gitlab.gnome.org/GNOME/mutter/-/issues/2390 was fixed in GNOME 43.

I think this can be closed now.

cc @GeorgesStavracas
